src/compiler/program.ts

```diff
@@ -2103,4 +2103,28 @@ namespace ts {
             return options.allowJs ? undefined : Diagnostics.Module_0_was_resolved_to_1_but_allowJs_is_not_set;
         }
     }
+
+    interface FileMap<T> {
+        get(fileName: Path): T;
+        set(fileName: Path, value: T): void;
+        has(fileName: Path): boolean;
+    }
+
+    function createFileMap<T>(keyMapper: (key: string) => string): FileMap<T> {

```

Comment:
I had a closer look at all the uses and couldn't answer this question. They were either using `keyMapper` redundantly (meaning, they were canonicalizing paths multiple times) or using it to convert things to lower case. It's simpler to just remove `createFileMap` entirely.